### ISSUE TITLE
TVM support

### ISSUE THREAD

**Issue by FrozenGene** [lutzroeder/netron#199]
Any plan to support NNVM / TVM ? which is very similar with MXNet. See NNVM / TVM graph json specification: https://docs.tvm.ai/dev/nnvm_json_spec.html

I could open it with *-symbol.json filename like MXNet. If you could support it more, will be nicer. For example, coloring. Currently, I open it , which just black / white.

![image](https://user-images.githubusercontent.com/7287321/49274244-b0a8ab00-f4b2-11e8-8495-89d236447e6e.png)

**Comment by lutzroeder** [comment]
Is the format the same as MXNet or are they just similar by accident? Is there a model zoo for TVM? From the description it seems to be not possible to detect if a `json` file is TVM or MXNet so that are the same. If that's the case feel free to submit a PR with improvements.

**Comment by lutzroeder** [comment]
@FrozenGene can you please work with the TVM team and provide:
* A model zoo or official reference files.
* A clear description or spec if TVM files are a special case of MXNet or a separate format. 
* How to detect if a file is TVM file. Are all TVM files named `-tvm.json` and did you change the TVM code to make sure this is the case. If not, are you planing to fix TVM to make it possible to easily detect the format?
* Why is the TVM format different from MXNet instead of just being compliant with MXNet?
* Most TVM operators seem to be fused so they wouldn't match with existing colors for known layers. How is this expected to work? Is there a spec listing all possible operators?

**Comment by FrozenGene** [comment]
Thanks for quick response. For TVM or MXNet, we could not control the name of .json file, which will be decided by user. As you did before, you assume the user will provide one .json file ends with -symbol.json, which will be MXNet model. So, I follow your way and think of one way that we can assume TVM user will provide the .json file ends with -tvm.json.

Then I answer your questions.
1. TVM haven’t one model zoo, I will talk about TVM team about this thing

2.TVM’ graph (called NNVM graph) is a plain json, MXNet also levarge NNVM, so MXNet’s model is also one NNVM graph. So they have many common places and format. But TVM has its unique ops like you have added.

3. Answered before.

4. TVM is a inference framework (can accept CoreML model, tensorflow model. MXNet model and so on), which is different with MXNet. MXNet focus on training, not inference. But they are both from DMLC community.

5. All TVM Ops will start with fuse_, because TVM will do graph fusion optimization. So if I have one convolution op, TVM graph will have fuse_conv2d, if I have batchnorm after it, I will have broadcast_add_broadcast_mul (TVM will do one optimization called SimplyInference and split batchnorm into add and multiply for precomuting), then I will have fuse_conv2d_broadcast_add_broadcast_mul. I know it have some strange places for parsing, but I will advice we parse convolution firstly, because in TVM, we will do graph fusion optimization defaultly and convolution will be much likely be the begining, because convolution can not be fused by other operators. So we can assume TVM op’s name will start fuse_, then if we have conv2d next, we mark it convolution’s color, then we can have one good place to start.

**Comment by lutzroeder** [comment]
It would help the have a few sample files available. I was not able to find a single TVM `json` file online.

**Comment by lutzroeder** [comment]
It would help to have a few sample files available. Online search came back empty.

If a file includes `tvm_op`, can it be assumed that all nodes will be of type `tvm_op` and hence the file must be a TVM file and should be using different metadata?

**Comment by FrozenGene** [comment]
> It would help to have a few sample files available. Online search came back empty.
> 
> If a file includes `tvm_op`, can it be assumed that all nodes will be of type `tvm_op` and hence the file must be a TVM file and should be using different metadata?

TVM Only contain `tvm_op` (which is used for normal operators, like conv2d) or `null` (which is used for input or params like weights), so if a file includes `tvm_op`, you can assume it is TVM.

I can provide one model file mobilenet v1 model. see zip compressed file here:

[mobilenet-v1-tvm.json.zip](https://github.com/lutzroeder/netron/files/2636924/mobilenet-v1-tvm.json.zip)

If you need more models, I can help to construct it.

**Comment by lutzroeder** [comment]
Duplicate of #1342
